gotham/src/state/request_id.rs

```diff
@@ -22,15 +21,14 @@ pub(super) struct RequestId {
 /// that a value for `RequestId` is always available.
 pub(crate) fn set_request_id<'a>(state: &'a mut State) -> &'a str {
     if !state.has::<RequestId>() {
-        let request_id = match Headers::borrow_from(state).get::<XRequestId>() {
+        let request_id = match HeaderMap::borrow_from(state).get("X-Request-ID") {
```

Comment:
Goes back to the comment further up about utilities for headers we "recognise" officially.